### PR TITLE
Remove YUI from the keyboard shortcuts.

### LINKED ISSUE #2634: Remove YUI from keyboard shortcuts in app.js
The keyboard shortcut code is all inter-twined with YUI DOM stuff. We want to remove YUI slowly, so refactor away.

### PR DESCRIPTION
**Pull request by huwshimi**
Fixes #2634.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- check that you can still open the shortcuts/settings panels
- check that other shortcuts still work

File: jujugui/static/gui/src/app/app.js [hunk 1]
```diff
@@ -111,44 +111,45 @@ YUI.add('juju-gui', function(Y) {
...
+                    if (node.get('checked')) {
+                      localStorage.setItem(node.getAttribute('name'), true);
+                    } else {
+                      // otherwise unset it from the localStorage.
+                      localStorage.removeItem(node.getAttribute('name'));
+                    }
                   } else {
-                    // otherwise unset it from the localStorage.
-                    localStorage.removeItem(node.getAttribute('name'));
+                    localStorage.setItem(
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/app.js]: It looks like the diff gets really wacky here, but all I did was indent the code inside the event listener.

File: jujugui/static/gui/src/app/app.js [hunk 2]
```diff
@@ -244,33 +245,6 @@ YUI.add('juju-gui', function(Y) {
...
     /**
-     * Data driven behaviors.
-     *
-     * Placeholder for real behaviors associated with DOM Node data-*
-     * attributes.
-     *
-     * @attribute behaviors
-     */
-    behaviors: {
-      timestamp: {
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/app.js]: I don't think we've used this for a long long time.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7066/

File: jujugui/static/gui/src/app/app.js [hunk 3]
```diff
@@ -2778,9 +2738,9 @@ YUI.add('juju-gui', function(Y) {
         // The current environment is not MAAS.
         return;
       }
-      var maasContainer = Y.one('#maas-server');
-      maasContainer.one('a').set('href', maasServer);
-      maasContainer.show();
+      var maasContainer = document.querySelector('#maas-server');
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/app.js]: As a drive by I've also removed the other `Y.one` instances in app.js.

File: jujugui/templates/index.html.go [hunk 4]
```diff
@@ -160,8 +160,8 @@ with this program.  If not, see <http://www.gnu.org/licenses/>.
       </div>
       <div id="viewport">
         <div id="content">
-            <div id="shortcut-help" class="modal" style="display:none"></div>
-            <div id="shortcut-settings" class="modal modal--narrow" style="display: none"></div>
+            <div id="shortcut-help" class="modal hidden"></div>
+            <div id="shortcut-settings" class="modal modal--narrow hiden"></div>
```
Comment (kadams54) [inline jujugui/templates/index.html.go]: Typo: 'hidden'.

**Comment by kadams54** [comment]
QA is OK. Thanks @huwshimi!

**Comment by kadams54** [comment]
Fixed the typo. :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7075/

**Pull request merged**